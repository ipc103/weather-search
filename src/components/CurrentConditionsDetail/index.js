import React from 'react'

const CurrentConditionsDetail = ({conditions}) => {
  const { city, weather, temp, message } = conditions

  if ( message ) {
    return <p>{message}</p>
  }

  return (
    <div>
      <h1>Current Conditions in {city}</h1>
      <p>The weather is {weather} with a temperature of {temp}</p>
    </div>
  )
}

CurrentConditionsDetail.defaultProps = {
  conditions: {
    message: 'No weather info found'
  }
}

export default CurrentConditionsDetail
