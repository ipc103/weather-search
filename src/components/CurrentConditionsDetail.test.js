import React from 'react'
import {shallow} from 'enzyme'
import CurrentConditionsDetail from './CurrentConditionsDetail'

describe('CurrentConditionsDetail', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<CurrentConditionsDetail conditions={ {} }/>)
    expect(wrapper).toMatchSnapshot()
  });
})
