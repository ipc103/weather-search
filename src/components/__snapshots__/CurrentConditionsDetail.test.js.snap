// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentConditionsDetail renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h1>
        Current Conditions in 
    </h1>
    <p>
        The weather is 
         with a temperature of 
    </p>
</div>,
  "nodes": Array [
    <div>
      <h1>
            Current Conditions in 
      </h1>
      <p>
            The weather is 
             with a temperature of 
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Unknown
        conditions={Object {}}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "conditions": Object {},
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h1>
                    Current Conditions in 
          </h1>
          <p>
                    The weather is 
                     with a temperature of 
          </p>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h1>
                    Current Conditions in 
          </h1>
          <p>
                    The weather is 
                     with a temperature of 
          </p>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Unknown
    conditions={Object {}}
/>,
}
`;
